File: Classes/PPHilightableButton.m
//
//  PPHilightableButton.m
//  PPMacho
//
//  Created by C.W. Betts on 12/21/12.
//
//

#import "PPHilightableButton.h"

@implementation PPHilightableButton

- (id)initWithFrame:(NSRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}

- (void)setState:(NSInteger)value
{
	[super setState:value];
	
	[self setNeedsDisplay:YES];
}

- (void)drawRect:(NSRect)dirtyRect
{
    [super drawRect:dirtyRect];
	if ([self state] == NSOnState) {
		NSColor *grayOverlay = [NSColor colorWithDeviceWhite:0.2 alpha:0.5];
		[grayOverlay set];
		//[grayOverlay drawSwatchInRect:dirtyRect];
		[[NSBezierPath bezierPathWithRect:dirtyRect] fill];
		
	}
}

@end

File: Shared/PPHilightableButton.swift
//
//  PPHilightableButton.swift
//  PPMacho
//
//  Created by C.W. Betts on 8/6/14.
//
//

import Cocoa

class PPHilightableButton: NSButton {

    init(frame: NSRect) {
        super.init(frame: frame)
        // Initialization code here.
    }
	
	func setState(value: Int) {
		super.state = value
		
		self.needsDisplay = true;
	}

    override func drawRect(dirtyRect: NSRect) {
        super.drawRect(dirtyRect)
		if (self.state == NSOnState) {
			var grayOverlay = NSColor(deviceWhite:0.2, alpha:0.5);
			grayOverlay.set()
			//[grayOverlay drawSwatchInRect:dirtyRect];
			var bezierPath = NSBezierPath(ovalInRect:dirtyRect)
			bezierPath.fill()
		}

        // Drawing code here.
    }
    
}
